File: backend/api/src/main/java/kpn/api/common/route/BaseRouteSegment.java
package kpn.api.common.route;

import kpn.api.common.Bounds;

import com.google.common.collect.ImmutableList;

public record BaseRouteSegment(
  Long id,
  Long startNodeId,
  Long endNodeId,
  Long meters,
  Bounds bounds,
  ImmutableList<Long> elementIds
) {}

File: backend/api/src/main/scala/kpn/api/common/route/BaseRouteSegment.scala
package kpn.api.common.route

import kpn.api.common.Bounds

case class BaseRouteSegment(
  id: Long,
  startNodeId: Long,
  endNodeId: Long,
  meters: Long,
  bounds: Bounds,
  elementIds: Seq[Long]
)
